termutil/term_aware_reader.go

```diff
@@ -0,0 +1,32 @@
+package termutil
+
+import (
+	"io"
+	"os"
+	"sync"
+
+	"golang.org/x/term"
+)
+
+type TermAwareReader struct {
```

Comment:
Like @AlexisMontagne, didn't find a better name... Idea are welcome

Reply:
`IsTermReader` ? Kind of hard to name that thing since it's will not read if the sub reader is a term. 
Maybe `NoTermReader` :sweat_smile: